tests/generics/e2e.test.ts

```diff
@@ -0,0 +1,147 @@
+import { expect } from 'chai';
+import { assertTCFail, assertPrint } from '../asserts.test';
+import {typeCheck} from "../helpers.test";
+
+describe('e2e tests to check generics', () => {
+    const prog0 = `
+    L = TypeVar('L')
+    R = TypeVar('R')
+
+    class Pair(Generic[L, R]):
+        left: L = __ZERO__
+        right: R = __ZERO__
```

Comment:
Test case to try – define `swap` here that produces a `Pair[R, L]`, and related constructions.

Reply:
Done!